modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/StoriesFragment.kt

```diff
@@ -39,16 +50,42 @@ class StoriesFragment : BaseAppCompatDialogFragment() {
             setContent {
                 AppTheme(theme.activeTheme) {
                     setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+                    analyticsTracker.track(AnalyticsEvent.END_OF_YEAR_STORIES_SHOWN, mapOf(AnalyticsProp.source to source.value))
                     StoriesPage(
                         viewModel = viewModel,
-                        onCloseClicked = { dismiss() },
+                        onCloseClicked = {
+                            analyticsTracker.track(AnalyticsEvent.END_OF_YEAR_STORIES_DISMISSED, AnalyticsProp.StoriesDismissed.closeButton)
+                            dismiss()
+                        },
                     )
                 }
             }
         }
     }
 
+    private object AnalyticsProp {
+        const val source = "source"
```

Comment:
Instead of exposing the `source` String key from the `AnalyticsProp` object, I'd be tempted to have a function that handles the `StoriesSource`:

``` kotlin
private object AnalyticsProp {
  private const val source = "source"
  fun storiesShown(storiesSource: StoriesSource) = mapOf(source to storiesSource.value)
  // ...
}
```

What you have is fine though, just sharing as food for thought.

Reply:
I like the suggestion, applied in 9e9749b.